usr/src/uts/common/brand/lx/os/lx_brand.c

```diff
@@ -2361,9 +2361,15 @@ lx_elfexec(struct vnode *vp, struct execa *uap, struct uarg *args,
 	 * Save off the important properties of the lx executable. The brand
 	 * library will ask us for this data later, when it is ready to set
 	 * things up for the lx executable.
+	 * For statically linked executable, use vaddr from Phdr.
 	 */
-	edp.ed_phdr = (uphdr_vaddr == -1) ? voffset + ehdr.e_phoff :
-	    voffset + uphdr_vaddr;
+	if (uphdr_vaddr == -1) {
+		edp.ed_phdr = voffset + ehdr.e_phoff;
+	} else if (ehdr.e_type == ET_EXEC && interp == NULL) {
+		edp.ed_phdr = uphdr_vaddr;
+	} else
+		edp.ed_phdr = voffset + uphdr_vaddr;
```

Comment:
You should be bracing this last statement like the middle ones as well.  ALSO, a comment about which case this is as well.

Reply:
Ok. Will correct.

Reply:
ping

Reply:
Updated. Just didn't add comment for the last condition since that comes from Top comment.

Reply:
I don't see any Top comment that explains this (or the `voffset + ehdr.e_phoff;` above it), or do you mean the Top of the function?

Reply:
Thank you!